webapp-django/crashstats/crashstats/jinja2/crashstats/report_index.html

```diff
@@ -691,44 +691,46 @@ <h2>{{ report.product }} {{ report.version }} Crash Report [@ {{ report.signatur
               </tbody>
             </table>
 
-            {% if (request.user.has_perm('crashstats.view_pii') or your_crash) and report.breadcrumbs %}
-              <div id="breadcrumbs">
-                <h2>Breadcrumbs</h2>
-                <p>
-                  {{ protected_warning(your_crash) }} -
-                  {% if report.breadcrumbs|length > 10 %}
-                    Latest 10 breadcrumbs (out of {{ report.breadcrumbs|length }})
-                  {% else %}
-                    All {{ report.breadcrumbs|length }} breadcrumbs
+            {% if java_exception_stacks %}
+              <div id="frames">
+                {% for stacktrace in java_exception_stacks %}
+                  {% set stacktrace = stacktrace.stacktrace %}
+                  <h2>Exception: {{ stacktrace.module }}.{{ stacktrace.type }}</h2>
+
+                  {% if stacktrace.value %}
+                    <h3>
+                      Message: {{ stacktrace.value }}
+                    </h3>
+                    <p>
+                      <img src="{{ static('img/3rdparty/silk/exclamation.png') }}" alt="protected" title="Protected Data" />
+                      Note: Unredacted exception messages are protected data.
+                      <a href="{{ url('documentation:protected_data_access') }}">Protected data policy.</a>
+                    </p>
```

Comment:
This was tricky. The only part of the stacktrace that's protected data is the message. I hope the way I've got it structured and the messaging I'm using gets that across.